### PR TITLE
Tests for navigating window.opener across origins

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

File: html/browsers/windows/opener-cross-origin-manual.sub.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<ol>
+ <li><p>Clicking this link must navigate this page to example.com:
+ <a href=//www1.{{location[host]}}/html/browsers/windows/support-opener.html target=_blank>test</a>
```
Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: `{{location[port]}}`?

Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: `{{domains[www1]}}`?

File: html/browsers/windows/opener-cross-origin-manual.sub.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<ol>
+ <li><p>Clicking this link must navigate this page to example.com:
```
Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: 1. `example.com`?
2. Why "this page" and not "a new window"?

Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: Ah, sorry, now I see the `support-opener.html` below.

File: html/browsers/windows/opener-cross-origin-manual.sub.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+ <a href=//www1.{{location[host]}}/html/browsers/windows/support-opener.html target=_blank>test</a>
+
+ <li><p>Clicking this link must open a new browsing context that is empty:
+ <a rel=noreferrer href=//www1.{{location[host]}}/html/browsers/windows/support-opener.html target=_blank>test</a>
+
+ <li><p>Clicking this link must navigate this page to example.com:
+ <a href=//www1.{{location[host]}}/html/browsers/windows/support-embed-opener.sub.html target=_blank>test</a>
```
Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: This doesn't look like it should work. That is, if you open `www1.whatever`, which frames `something-else.whatever`, then my intuition is that the cross-origin frame shouldn't be able to reach up into it's parent's opener, or close it's parent. Does that work in Firefox? I guess I'll have to run this in Chrome to see what it does.

Comment (annevk) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: It works in Firefox. In Chrome it seems you can simply first navigate the parent and then navigate the opener so I'm not sure why Chrome restricts this.

Comment (mikewest) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: In Chrome, the navigation fails, but the `close()` doesn't. I'm not sure if I agree with that behavior.

Comment (annevk) [inline html/browsers/windows/opener-cross-origin-manual.sub.html]: If you could figure out the rationale of what Chrome is doing here that would help @bzbarsky and I.

**Comment by gsnedders** [comment]
```
15:01 < aryehgregor> Um, I might have done something wrong by mistake.
15:01 < aryehgregor> I think I pushed to the wrong repo?
```

**Comment by bzbarsky** [comment]
@ayg Is this really checked in for real now?  Or is this like the previous time?

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/2356/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/212039352)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 2f58168dd9ea56b160dab4ce1a910b78829b5564
Using browser at version BuildID 20170314110401; SourceStamp 6d38ad302429c98115c354d643e81987ecec5d3c
Starting 10 test iterations
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/212039353)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 2f58168dd9ea56b160dab4ce1a910b78829b5564
Using browser at version 58.0.3029.19 dev
Starting 10 test iterations
No tests run.

**Comment by mikewest** [comment]
Tests look fine to me: the first two pass in Chrome, the last does not because of the logic in https://cs.chromium.org/chromium/src/third_party/WebKit/Source/core/frame/Frame.cpp?rcl=56b1ac1be5b34735541c864d6de5b61e3091258d&l=347 which enforces a direct relationship between the window initiating the navigation and the window being navigated (same-origin, `parent`, or `opener`).

Given the reference to @abarth, et al's 2008 paper in that part of the code, I assume it simply predates the current definition in HTML. As you note, the frame can simply navigate its parent to a page that can navigate `opener`, so we should probably change Blink to align with Gecko.

**Comment by annevk** [comment]
Cool, will you file a bug against Chrome?

**Comment by annevk** [comment]
I should maybe also get review from @lyzadanger since she has been cleaning up this space.

**Comment by mikewest** [comment]
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=702545.

**Comment by lyzadanger** [comment]
@annevk @mikewest My browsing context texts have been primarily focused on same-origin stuff, but, in squinting at https://html.spec.whatwg.org/multipage/browsers.html#crossoriginproperties-(-o-), it does seem to me that FireFox's behavior here is correct as has been asserted: it (Firefox, correctly) is allowing x-origin access to `location` and its `href` property.

Also consistent is that `parent.opener.open()` does _not_ work in FF, as `open` is not a property that can be accessed x-origin (I tried that out for my own curiosity in a few browsers).

I ran into something yesterday with FF's treatment of a (same-origin) `opener` when a `parent` browsing context is reused via `window.open` targeting, so this thread piqued my interest as it involved `opener` and `parent`—but they are not related.

**Comment by annevk** [comment]
Thank you both!

**Comment by bzbarsky** [comment]
@annevk Do you know whether there's a Gecko bug tracking these tests?  Or does Firefox pass them already?  There weren't any test results above for some reason.  :(

**Comment by annevk** [comment]
@bzbarsky these are manual tests and Firefox passes them already.

**Pull request merged**